examples/api/test/widgets/framework/error_widget.0_test.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/framework/error_widget.0.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('ErrorWidget is displayed in debug mode', (WidgetTester tester) async {
+    final ErrorWidgetBuilder oldBuilder = ErrorWidget.builder;
+    ErrorWidget.builder = (FlutterErrorDetails details) {
+       return ErrorWidget(details.exception);
+    };
+    await tester.pumpWidget(const example.ErrorWidgetExampleApp());
+
+    expect(find.widgetWithText(AppBar, 'ErrorWidget Sample'), findsOne);
+
+    await tester.tap(find.widgetWithText(TextButton, 'Error Prone'));
+    await tester.pump();
+
+    expectLater(tester.takeException(), isInstanceOf<Exception>());
+
+    final Finder errorWidget = find.byType(ErrorWidget);
+    expect(errorWidget, findsOneWidget);
+    final ErrorWidget error = tester.firstWidget(errorWidget);
+    expect(error.message, 'Exception: oh no, an error');
+
+    ErrorWidget.builder = oldBuilder;
```

Comment:
What this does at the end of the test? Is this ncessary?

Reply:
yes, without it the test ends up failing by throwing this error

> ══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following assertion was thrown:
The value of ErrorWidget.builder was changed by the test.

Reply:
Could you please add a comment for this? It's not normal to end a test with additional steps after the expectations.

Reply:
sure, Does it look okay like this ? @TahaTesser 
```suggestion
 // Restore original ErrorWidget.builder otherwise FLUTTER TEST FRAMEWORK will throw exception
 ErrorWidget.builder = oldBuilder;
```

Reply:
// Restore the `ErrorWidget` builder to conclude the test.